src/Http/Routing/src/DefaultLinkGenerator.cs

```diff
@@ -336,137 +336,92 @@ public void Dispose()
-                LogLevel.Debug,
-                EventIds.TemplateFailedRequiredValues,
-                "Failed to process the template {Template} for {Endpoint}. " +
-                "A required route value is missing, or has a different value from the required default values. " +
-                "Supplied ambient values {AmbientValues} and {Values} with default values {Defaults}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, string, string, IRouteConstraint, string, string, Exception> _templateFailedConstraint = LoggerMessage.Define<string, string, IRouteConstraint, string, string>(
-                LogLevel.Debug,
-                EventIds.TemplateFailedConstraint,
-                "Failed to process the template {Template} for {Endpoint}. " +
-                "The constraint {Constraint} for parameter {ParameterName} failed with values {Values}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, string, string, string, Exception> _templateFailedExpansion = LoggerMessage.Define<string, string, string>(
-                LogLevel.Debug,
-                EventIds.TemplateFailedExpansion,
-                "Failed to process the template {Template} for {Endpoint}. " +
-                "The failure occurred while expanding the template with values {Values} " +
-                "This is usually due to a missing or empty value in a complex segment",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, IEnumerable<string>, string, Exception> _linkGenerationSucceeded = LoggerMessage.Define<IEnumerable<string>, string>(
-                LogLevel.Debug,
-                EventIds.LinkGenerationSucceeded,
-                "Link generation succeeded for endpoints {Endpoints} with result {URI}",
-                skipEnabledCheck: true);
-
-            private static readonly Action<ILogger, IEnumerable<string>, Exception> _linkGenerationFailed = LoggerMessage.Define<IEnumerable<string>>(
-                LogLevel.Debug,
-                EventIds.LinkGenerationFailed,
-                "Link generation failed for endpoints {Endpoints}",
-                skipEnabledCheck: true);
-
-            public static void EndpointsFound(ILogger logger, object address, IEnumerable<Endpoint> endpoints)
-            {
-                // Checking level again to avoid allocation on the common path
-                if (logger.IsEnabled(LogLevel.Debug))
-                {
-                    _endpointsFound(logger, endpoints.Select(e => e.DisplayName), address, null);
-                }
-            }
+            [LoggerMessage(EventId = 100, EventName = "EndpointsFound", Level = LogLevel.Debug, Message = "Found the endpoints {Endpoints} for address {Address}")]
```

Comment:
@maryamariyan / @eerhardt could you have a look at this and let me know if I'm doing this correctly?

FYI @JamesNK / @davidfowl

Reply:
The previous `EndpointsFound` method would check whether logging was enabled and then use LINQ to get endpoint names. I believe that is missing post your changes.

Follow the pattern you used with TemplateFailedRequiredValues?